### ISSUE TITLE
SteamVR crashes when being launched/opened

### ISSUE THREAD

**Issue by DavisNT** [JackD83/ALVR#142]
**Issue** 

1. Sometimes (or always) SteamVR immediately closes when being launched/opened from ALVR. 
1. In `C:\Program Files (x86)\Steam\logs\vrserver.txt` something like `Return to watchdog timeout of XXXX.YYYYYY seconds exceeded. Quitting everything.` is logged. 
1. In `C:\Program Files (x86)\Steam\dumps` files named `crash_vrserver.exe_YYYYMMDDHHMMSS_X.dmp` are created **some** of the times when SteamVR closes.

**Solution**

- **If you can open SteamVR:** Open SteamVR settings, open _Startup / Shutdown_ tab and set _Exit SteamVR after_ to _NEVER_. 
- **If you CANNOT open SteamVR (if it always closes):** Edit file `C:\Program Files (x86)\Steam\config\steamvr.vrsettings` and remove these lines: 

       "power" : {
          "returnToWatchdogTimeout" : 3600
       },

P.S. Looks like this bug has appeared in recent version(s) of SteamVR. 
P.P.S. I am not sure whether this could somehow be fixed in ALVR.

**Comment by ratozumbi** [comment]
always SteamVR immediately closes when being launched/opened from ALVR

I tried your solution by editing steamvr.vrsettings but the problem persists

**Comment by DavisNT** [comment]
@ratozumbi Remove the line `"returnToWatchdogTimeout" : 3600,` from your config.

**Comment by ratozumbi** [comment]
> @ratozumbi Remove the line `"returnToWatchdogTimeout" : 3600,` from your config.

Done that and the problem persist.

I was able to sustain SteamVR open by altering a configuration on ALVR: on "other" tab mark checkbox "Launch only SteamVR without Steam"
but...
ALVR crashes with error:
`Failed to initialize CEncoder. All VideoEncoder are not available. VCE: AMF Error 1. g_AMFFactory.Init(), NVENC: Exception:Failed to get CUDA device. `

**Comment by ratozumbi** [comment]
TL;DR:
My laptop is still a mystery, but PC is doing fine.

I was trying to run ALVR server on my laptop. It uses and integrated graphics card to run desktop stuff, but the GTX 960M kicks in when playing games. This laptop is known to run StemVR with a Windows Mixed Reality headset (lenovo explorer).
When I tried on my PC (GTX 970) ALVR worked first time with no issued. Except maybe that the controller looks like is delayed, as if I was underwater. But I guess that can be fixed trough some configuration.

**Comment by DavisNT** [comment]
@ratozumbi Do you get the log entry when SteamVR closes immediately after launching? Are the dump (.dmp) files created? 
Are you sure Steam and SteamVR were fully closed when editing the `steamvr.vrsettings`?

> I was able to sustain SteamVR open by altering a configuration on ALVR: on "other" tab mark checkbox "Launch only SteamVR without Steam" 

In my case it was also possible to work-around the issue by using this setting. 

> but...
> ALVR crashes with error:
> `Failed to initialize CEncoder. All VideoEncoder are not available. VCE: AMF Error 1. 

This error message looks like some other issue (maybe laptop using only integrated graphics ). 

P.S. Please always be specific (currently it is not clear on which computer you have what issue(s)).

**Comment by ETHANATER50** [comment]
SteamVR is crashing for me every time. ALVR reports:
 Unhandled Exception.
ExceptionCode=C0000005
Address=00007FFD94105D04 (00007FFDACE40000 + FFFFFFFFE72C5D04)

then steam tells me that "A key component of SteamVR isn't working properly (301)

it'll boot up if I'm not running ALVR, but then I can't actually play it. The line you suggested deleting doesn't actually exist in my vrsettings file. I'm on SteamVR version 1.10.32 if that helps

**Comment by DavisNT** [comment]
@ETHANATER50 Is the `Return to watchdog timeout`... message logged (see point 2 of the issue rescription)? Is there anything else notable in the log? 
Can you try an experiment - rename `steamvr.vrsettings` file and check whether SteamVR launches then (and check whether it works without changing any optional settings)?

**Comment by ACollectionOfAtoms** [comment]
Funnily enough I had my on-board video card disabled and kept getting the `Failed to initialize CEncoder. A` error (latest version of client and server). I then turned it back on and everything worked! This is a running with a https://www.amazon.com/gp/product/B07QQB6DC1/ref=ppx_yo_dt_b_asin_title_o07_s00?ie=UTF8&psc=1

**Comment by ratozumbi** [comment]
I will just link this as it looks like the only other post related to this issue.
I will try this solution later and see if it works
https://www.reddit.com/r/OculusQuest/comments/epek90/alvr_not_working_again/

**Comment by DavisNT** [comment]
@ACollectionOfAtoms @ratozumbi This issue is about completely different thing (**not** the `Failed to initialize CEncoder.` problem). 
Please create a separate GitHub issue for it or discuss it in #130 if you think it is the same issue!
